apps/challenges/aws_utils.py

```diff
@@ -0,0 +1,301 @@
+                challenge.workers = 1
+                challenge.save()
+                message = {"Success": "Service created succesfully."}
+                return message
+            return "Creation for challenge {}.".format(challenge.pk)
+        except ClientError as e:
+                return e.response["Error"]
+    return "Worker service for challenge {} already exists. Please scale, stop or delete.".format(challenge.pk)
+
+
+# Called by the service_manager.
+def update_service_by_challenge_pk(client, challenge, num_of_tasks):
+    queue_name = challenge.queue
+    service_name = "{}_service".format(queue_name)
+    task_def_arn = challenge.task_def_arn
+
+    definition = update_service_args.format(service_name=service_name, task_def_arn=task_def_arn)
+    definition = eval(definition)
+
+    try:
+        response = client.update_service(**definition)
+        # To verify if the service has been updated on AWS.
+        if (response["ResponseMetadata"]["HTTPStatusCode"] == 200) and (response["service"]["serviceName"] == service_name):
+            challenge.workers = num_of_tasks
+            challenge.save()
+            message = {"Success": "Service updated succesfully."}
+            return message
+        return "Update failed for challenge {}.".format(challenge.pk)
+    except ClientError as e:
+        return e.response['Error']
+
+"""
+Called by: Start, Stop & Scale methods for multiple workers. (Below)
+
+This method determines if the challenge is new or not, and accordingly
+calls the create or update custom method.
+"""
+
+
+def service_manager(client, challenge, num_of_tasks=None):
+    if challenge.workers is not None:
+        response = update_service_by_challenge_pk(client, challenge, num_of_tasks)
+        return response
+    else:
+        client_token = client_token_generator()
+        response = create_service_by_challenge_pk(client, challenge, client_token)
+        return response
+
+"""
+The 3 functions below are used directly in the admin action methods and the response returned by
+these are displayed as success or error messages on the admin page
+
+Called by: their respective admin action methods in apps/challenges/admin.py.
+"""
+
+
+def start_workers(queryset):
+    ecs = get_boto3_client("ecs", aws_keys)
+    count = 0   # Num of succesful starts.
+    for challenge in queryset:
+        if (challenge.workers == 0) or (challenge.workers is None):  # To verify if the challenge is inactive or new.
+            response = service_manager(client=ecs, challenge=challenge, num_of_tasks=1)
+            if "Success" not in response:
+                # This response is formatted in the admin page to give 2 messages
+                # 1)Num of succesful starts before error. 2)The error if any for the first failure.
+                return {"count": count, "message": response}
+            count += 1
+        else:
+            response = {"Error": "Please select only inactive workers."}
```

Comment:
Shouldn't the message be `Please select only inactive challenges.`

Reply:
Yeah, that's more accurate. Thanks for pointing out.